pkg/resources/object_renaming_lists_and_sets.go

```diff
@@ -0,0 +1,518 @@
+	list := collections.Map(ObjectRenamingDatabaseInstance.List, func(t objectRenamingDatabaseListItem) map[string]any {
+		return map[string]any{
+			"name":   t.Name,
+			"string": t.String,
+			"int":    t.Int,
+		}
+	})
+	if err := d.Set("list", list); err != nil {
+		return diag.FromErr(err)
+	}
+
+	orderedList := make([]map[string]any, len(ObjectRenamingDatabaseInstance.OrderedList))
+	for index, item := range ObjectRenamingDatabaseInstance.OrderedList {
+		orderedList[index] = map[string]any{
+			"name":  item.Name,
+			"order": strconv.Itoa(index),
+		}
+	}
+	if err := d.Set("ordered_list", orderedList); err != nil {
+		return diag.FromErr(err)
+	}
+
+	manuallyOrderedList := make([]map[string]any, len(ObjectRenamingDatabaseInstance.ManuallyOrderedList))
+	for index, item := range ObjectRenamingDatabaseInstance.ManuallyOrderedList {
+		manuallyOrderedList[index] = map[string]any{
+			"name":  item.Name,
+			"order": index + 1, // We start from 1, because there may be some implications with 0 and the fact it's treated as unset value in Terraform
+		}
+	}
+	if err := d.Set("manually_ordered_list", manuallyOrderedList); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func DeleteObjectRenamingListsAndSets(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	ObjectRenamingDatabaseInstance.List = nil
+	ObjectRenamingDatabaseInstance.OrderedList = nil
+	ObjectRenamingDatabaseInstance.ManuallyOrderedList = nil
+	d.SetId("")
+
+	return nil
+}
+
+func ignoreListOrderAfterFirstApply(parentKey string) schema.SchemaDiffSuppressFunc {
+	return func(key string, oldValue string, newValue string, d *schema.ResourceData) bool {
+		if strings.HasSuffix(key, ".#") {
+			return false
+		}
+
+		// Raw state is not null after first apply
+		if !d.GetRawState().IsNull() && oldValue != newValue {
+			// Parse item index from the key
+			keyParts := strings.Split(strings.TrimLeft(key, parentKey+"."), ".")
+			if len(keyParts) >= 2 {
+				index, err := strconv.Atoi(keyParts[0])
+				if err != nil {
+					log.Println("[DEBUG] Failed to convert list item index: ", err)
+					return false
+				}
+
+				// Get the hash of the whole item from config (because it represents new value)
+				newItemHash := d.GetRawConfig().AsValueMap()[parentKey].AsValueSlice()[index].Hash()
+
+				newItemWasAlreadyPresent := false
+
+				// Try to find the same hash in the state; if found, the new item was already present, and it only changed place in the list
+				for _, oldItem := range d.GetRawState().AsValueMap()[parentKey].AsValueSlice() {
+					// Matching hashes indicate the order changed, but the item stayed in the config, so suppress the change
+					if oldItem.Hash() == newItemHash {
+						newItemWasAlreadyPresent = true
+					}
+				}
+
+				oldItemIsStillPresent := false
+
+				// Sizes of config and state may not be the same
+				if len(d.GetRawState().AsValueMap()[parentKey].AsValueSlice()) > index {
```

Comment:
These functions are deeply nested - pls try using `collections.Map`, extracting some code and returning earlier.